File: clientApp/src/models/Ativo.model.ts
export class Ativo {
  public id: string;
  public codigo: string;
  public valor: number;
  public quantidadeNegociados: number;

  constructor(id: string, codigo: string, valor: number, quantidadeNegociados: number) {
      this.id = id;
      this.codigo = codigo;
      this.valor = valor;
      this.quantidadeNegociados = quantidadeNegociados;
  }
}

File: api/Desafio.Application/Models/Ativos/AtivoModel.cs
using System;

namespace Desafio.Application.Models.Ativos
{
    public class AtivoModel
    {
        public Guid Id { get; set; }
        public string Codigo { get; set; }
        public decimal Valor { get; set; }
        public int QuantidadeNegociados { get; set; }
    }
}